scylla.go

```diff
@@ -413,15 +413,14 @@ func (p *scyllaConnPicker) maybeReplaceWithLessBusyConnection(c *Conn) *Conn {
 		return c
 	}
 	alternative := p.leastBusyConn()
-	if alternative == nil || alternative.AvailableStreams()*120 > c.AvailableStreams()*100 {
-		return c
-	} else {
+	if alternative != nil && alternative.StreamsInUse()*100 >= c.StreamsInUse()*(100-c.session.cfg.HeavyLoadedSwitchConnectionPercentage) {
 		return alternative
 	}
+	return c
 }
 
 func isHeavyLoaded(c *Conn) bool {
-	return c.streams.NumStreams/2 > c.AvailableStreams()
+	return c.StreamsInUse() > c.session.cfg.HeavyLoadedConnectionThreshold
```

Comment:
This part was already configurable trough `MaxRequestsPerConn` parameter in ClusterConfig.
I think we should either stick with the old parameter or remove it. Having 2 will be confusing.
Also, currently this PR changes the default (previously it was 32768 / 2, now it is 512). Why this number and why should the default be changed?

Reply:
Having it in an old way do not allow user to switch early and allow to have lot's of requests in flight.
`MaxRequestsPerConn` used to control not only max number of requests in flight, but also a waterline after which connection is considered under high load.
This kind of coupling is never good.
Beside that, before this PR driver considered connection to be highly loaded only after reaching 16k requests in flight, which is a HUGE number, it is safe to say that this feature never worked before.

Having a separate configuration for a heavy loaded water line allows user to allow driver to have big number of in flight requests and at the same time to switch to underutilized connections early.